automations/python/workflows/generate_report.py

```diff
@@ -0,0 +1,51 @@
+"""Generate a completion report for the CI + CD workflow."""
+
+import json
+import os
+import sys
+from collections import defaultdict
+
+
+server_url = os.environ.get("SERVER_URL")
+repository = os.environ.get("REPOSITORY")
+run_id = os.environ.get("RUN_ID")
+
+jobs = ["emit-docs", "publish-images", "deploy-frontend", "deploy-api"]
+
+results = {}
+counts = defaultdict(lambda: 0)
```

Comment:
This can be replaced with the `int` type:
```suggestion
counts = defaultdict(int)
```